keel-sql/src/main/resources/db/changelog/20210226-generate-event-columns.yml

```diff
@@ -0,0 +1,80 @@
+databaseChangeLog:
+- changeSet:
+    id: generate-event-timestamp
+    author: fletch
+    changes:
+    - dropIndex:
+        tableName: event
+        indexName: event_uid_timestamp_idx
+    - dropIndex:
+        tableName: event
+        indexName: event_scope_ref_timestamp_idx
+    - dropIndex:
+        tableName: event
+        indexName: event_scope_application_timestamp_idx
+    - dropColumn:
+        tableName: event
+        columnName: ref
+    - dropColumn:
+        tableName: event
+        columnName: application
+    - dropColumn:
+        tableName: event
+        columnName: timestamp
+    - sql:
+        sql: |
+          alter table event
+          add column ref varchar(255) generated always as (json ->> '$.ref')
+          not null
+          after scope;
+    - sql:
+        sql: |
+          alter table event
+          add column application varchar(255) generated always as (json ->> '$.application')
+          not null
+          after ref;
+    - sql:
+        sql: |
+          update event
+          set json = json_set(
+            json,
+            '$.timestamp',
+            concat(substring(json_unquote(json->'$.timestamp'), 1, 19), '.000Z')
+          )
+          where length(json -> '$.timestamp') = 22;
+    - sql:
+        sql: |
+          alter table event
+          add column timestamp datetime(3) generated always as (str_to_date(json_unquote(json->'$.timestamp'), '%Y-%m-%dT%T.%fZ'))
```

Comment:
```suggestion
          add column timestamp datetime(3) generated always as (str_to_date(json->>'$.timestamp', '%Y-%m-%dT%T.%fZ'))
```

Reply:
huh, I didn't know that syntax. Nice.

Reply:
I learned it from your PR, haha! 😄 
https://github.com/spinnaker/keel/blob/c93a4ce8cea1f7106153e0853be51b6a1fa0723d/keel-sql/src/main/resources/db/changelog/20210226-generate-event-columns.yml#L27